core/src/autogluon/core/scheduler/mo_asha.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import copy
+import numpy as np
+
+from .hyperband import HyperbandScheduler, HyperbandBracketManager
+from .mo_asha_promotion import EPS_NET, NSGA_II
+from ..utils.default_arguments import filter_by_key
+from ..utils.mo_hbo_utils import prepare_sign_vector
+from ..utils.mo_hbo_utils import retrieve_pareto_front
+
+
+__all__ = ['MOASHAScheduler']
+
+logger = logging.getLogger(__name__)
+
+_ARGUMENT_KEYS = {
+    'objectives',
+}
+
+
+class MOASHAScheduler(HyperbandScheduler):
```

Comment:
I don't like this.
Either (a) you think that this scheduler is simply a variant of HyperbandScheduler with a different bracket manager, or (b) it is its own thing.
For (a), you insert code into the HyperbandScheduler constructor, but don't have a subclass.

For (b), you have a subclass, which has its own constructor code. If you want to avoid copy&paste, refactor parts of the constructor into private methods.

I'd prefer (b) to be honest, so the HyperbandScheduler code is not much modified.